### ISSUE TITLE
Icons missing after update to v1.12.0

### ISSUE THREAD

**Issue by mamoel666** [kalkih/mini-media-player#467]
This morning I updated to v1.12.0 via HACS.
Now the icons are missing.

Example config:

      - type: custom:mini-media-player
        icon: mdi:theater

I restarted HA and cleared browser cache.

Environment:
Home Assistant 2020.12.2 in Docker (Synology NAS).

**Comment by Krocko** [comment]
Same for me.

**Comment by agneevX** [comment]
For now, you can you can set icon to anything like `icon: s` and it reverts to the entity's icon.

**Comment by nepozs** [comment]
But icons do not show at all, last working version with working icons is 1.11.0 (so I've temporary downgraded mini media player)

**Comment by kineticscreen** [comment]
Yeah I can't quite work it out - it has something to do with when an icon is overidden in the front end UI via the entities page.

**Comment by koenadolfs** [comment]
Same here. I have icons overwritten at the entity level, but indeed do not show up. Weirdly, the effects seems to differ between devices.

**Comment by nepozs** [comment]
I don't use icon customization nor entity level icon override, so it does not look as cause…
![mini_media_player_no_entity_level_customization_icon_Screenshot_2021-01-13 Configuration - Home Assistant](https://user-images.githubusercontent.com/56661567/104397848-e2274800-554d-11eb-8183-acc89a02ed01.png)
![mini_media_player_no_customization_icon_Screenshot_2021-01-13 Configuration - Home Assistant](https://user-images.githubusercontent.com/56661567/104397852-e3587500-554d-11eb-9ec7-7fcbde4a540d.png)

**Comment by Drafteed** [comment]
Same issue

**Comment by iptvcld** [comment]
same issue for me; i ended up adding the icon mdi into the customize.yaml file

**Comment by cooperaj** [comment]
I have 6 mini media players on a tab of my UI. Each one is overridden in the ui using `icon: mdi:icon-name`. Only the first such icon on the page shows up. The others all show blank spaces.

If I remove the ui overrides then the default icon shows up just fine. As @iptvcld says above you can then use the customize.yml file to do what you want. I would rather not do that though.

**Comment by sasukebinbin** [comment]
Same here

**Comment by leranp** [comment]
Same here

**Comment by ranrinc** [comment]
Quick Fixed for those who want to do it manually while waiting for the next release. 

Edit your js on ./www/community/mini-media-player-bundle.js using your noteeditor

search for `return this.config.icon ? this.player.icon : ICON.DEFAULT;` and change it to `return this.config.icon ? this.player.icon || ICON.DEFAULT;` and you should have working mini-media-player

Enjoy

**Comment by nepozs** [comment]
@ranrinc It does not help, so as temporary fix downgrade to 1.11.0 stlli is better…

**Comment by ranrinc** [comment]
> @ranrinc It does not help, so as temporary fix downgrade to 1.11.0 stlli is better…

It's working for me...

![Screen Shot 2021-01-21 at 8 27 26 AM](https://user-images.githubusercontent.com/15801696/105262132-92d2be80-5bc2-11eb-9011-ad8c63fc0358.png)

![Screen Shot 2021-01-21 at 8 27 42 AM](https://user-images.githubusercontent.com/15801696/105262228-98c89f80-5bc2-11eb-9a7c-243f37652f42.png)

**Comment by akeslo** [comment]
> Quick Fixed for those who want to do it manually while waiting for the next release.
> 
> Edit your js on ./www/community/mini-media-player-bundle.js using your noteeditor
> 
> search for `return this.config.icon ? this.player.icon : ICON.DEFAULT;` and change it to `return this.config.icon ? this.player.icon || ICON.DEFAULT;` and you should have working mini-media-player
> 
> Enjoy

I made the change but how do I clear the JS cache so that this updates in the UI?

**Comment by ranrinc** [comment]
> > Quick Fixed for those who want to do it manually while waiting for the next release.
> > Edit your js on ./www/community/mini-media-player-bundle.js using your noteeditor
> > search for `return this.config.icon ? this.player.icon : ICON.DEFAULT;` and change it to `return this.config.icon ? this.player.icon || ICON.DEFAULT;` and you should have working mini-media-player
> > Enjoy
> 
> I made the change but how do I clear the JS cache so that this updates in the UI?

In my case refresh HA and you should see the implementations

**Comment by N36** [comment]
I got it working by just defining 'mdi:icon-name' under the icon tab in my HASSIO entities section.

**Comment by DavidFW1960** [comment]
> Edit your js on ./www/community/mini-media-player-bundle.js using your noteeditor
> 
> search for `return this.config.icon ? this.player.icon : ICON.DEFAULT;` and change it to `return this.config.icon ? this.player.icon || ICON.DEFAULT;` and you should have working mini-media-player

So I tried this hack but I do not find this text anywhere in the js
```
return this.config.icon ? this.player.icon : ICON.DEFAULT
```

I also installed the repo and looked for it in main.js file there and made a comparable change to source code but then it would not compile.



> I got it working by just defining 'mdi:icon-name' under the icon tab in my HASSIO entities section.

I don't understand what you mean with this? Which entity? what icon tab?

**Comment by N36** [comment]
![chrome_foYC5NVFzm](https://user-images.githubusercontent.com/4256908/107020055-affeb400-67a2-11eb-85fd-c56a53c4bd6d.png)
Just go into Configuration > Entities > Your-Entity. Then add the manual mdi icon as above. It will then appear in your lovelace UI.

**Comment by schteff** [comment]
Just to be clear. That is not a final solution, just a quick fix if you are in a hurry to see the icons. The real problem has to be fixed in mini-media-player and released in a new version.

**Comment by DavidFW1960** [comment]
Well I was attempting the quick fix and also fixing the card but the posted solution... well that text isn't found and if I edit the js file it won't compile

**Comment by tomlut** [comment]
Yeah my js file does not contain that text either.

**Comment by tomlut** [comment]
> 
> 
> @ranrinc It does not help, so as temporary fix downgrade to 1.11.0 stlli is better…

Rolling back to 1.11.0 made no difference (yes I cleared my browser cache).

Looks like Customize is the best solution for now.

**Comment by codylv7** [comment]
This was happening to me, so I installed the OneCleaner Chrome extension, which seems to have fixed my issue. Here are the steps I took, hopefully it works for others.

1. Install OneCleaner Chrome Extension from [here](https://chrome.google.com/webstore/detail/onecleaner/nnkejamhodoofikdollghgpiedafpgbi?hl=en-US)
2. Open the options for OneCleaner and enable "Clear when clicking on the icon in the toolbar", check all boxes under "Data to remove"
3. Browse to your HA site
4. Click the OneCleaner icon in toolbar
5. Set Time Period = Everything, do not use any other time period.
6. Check "Only Selected Websites" and check the box next to your HA site
7. Click "Clean Selected"
8. Close browser, then log back in to your HA site
